File: SendGrid_Email_Template.py
# Required Packages
import os
from dotenv import load_dotenv
from sendgrid import SendGridAPIClient
from sendgrid.helpers.mail import Mail

# Load environment variables
load_dotenv(dotenv_path="../.env")

# Import custom logger
from Error_Logger_Template import logger

# Call Synchronous Process
sg = SendGridAPIClient(os.environ.get('SENDGRID_API_KEY'))

message = Mail(
    to='recipient@email_provider.com',
    from_email='sender@company.com',
    subject='Sending Email with Sendgrid is Fun',
    plain_text_content='Test Message from Sendgrid API using Python',
    html_content='<strong>Test Message from Sendgrid API Python HTML Message</strong>'
)

try:
    response = sg.send(message)
    print("Email sent successfully to the recipient")
except Exception as e:
    logger.error(f"There was an error sending this email: {e}")

File: Email_Related/SendGrid_Email_Template.mjs
//Required Packages
import dotenv from 'dotenv'
dotenv.config({ path: "../.env" });
import sgMail from '@sendgrid/mail';


//JS Code
import logger from './Error_Logger_Template.mjs'


//Call Synchronous Process
sgMail.setApiKey(process.env.SENDGRID_API_KEY);

const msg  = {
    to:'recipient@email_provider.com',
    from: 'sender@company.com',
    subject: 'Sending Email with Sendgrid is Fun',
    text: 'Test Message from Sendgrid API using NodeJS',
    html: '<strong>Test Message from Sendgrid API NodeJS HTML Message</strong>',
}

sgMail
.send(msg)
.then(() => { 
    console.log("Email Eent Successfully to the Recipient"); 
})
.catch((err) => { 
    logger.error("There was an error sending this email" + (err||err.stack));
});